js/eventloop/eventloop.go

```diff
@@ -61,11 +61,17 @@ func (e *EventLoop) wakeup() {
 // This function *must* be called from within running on the event loop, but its result can be called from anywhere.
 func (e *EventLoop) RegisterCallback() func(func() error) {
 	e.lock.Lock()
+	var done bool
```

Comment:
hmm `done` is misleading here, something like this seems better:

```suggestion
	var callbackRegistered bool
```

Reply:
`callbackRegisted` is also not right - it is registered once this method returns. Maybe `called`

Reply:
:+1: for `called` or `callbackWasCalled`, if we want to be more descriptive